pkg/apis/pipeline/v1beta1/pipelinerun_validation.go

```diff
@@ -107,14 +110,79 @@ func (ps *PipelineRunSpec) Validate(ctx context.Context) (errs *apis.FieldError)
 			wsNames[ws.Name] = idx
 		}
 	}
-
 	for idx, trs := range ps.TaskRunSpecs {
 		errs = errs.Also(validateTaskRunSpec(ctx, trs).ViaIndex(idx).ViaField("taskRunSpecs"))
 	}
 
 	return errs
 }
 
+func (ps *PipelineRunSpec) validateInlineParameters(ctx context.Context) (errs *apis.FieldError) {
+	var paramSpec []ParamSpec
+	if ps.Params == nil || ps.PipelineSpec == nil {
+		return errs
+	}
+	for _, p := range ps.Params {
+		pSpec := ParamSpec{
+			Name:    p.Name,
+			Default: &p.Value,
+		}
+		paramSpec = append(paramSpec, pSpec)
+	}
+	for _, p := range ps.PipelineSpec.Params {
+		skip := false
```

Comment:
It seems to me like we can extract out some generic functions for the check here with the skip condition.